src/IO/WriteBufferFromS3.cpp

```diff
@@ -116,7 +116,14 @@ void WriteBufferFromS3::allocateBuffer()
 
 WriteBufferFromS3::~WriteBufferFromS3()
 {
-    finalize();
```

Comment:
Maybe it is better to put `finalize` call into caller to avoid suppressing errors?